test/core/room_e2e_test.dart

```diff
@@ -36,10 +36,16 @@ void main() {
     test('participant join', () async {
       expect(
         room.events.streamCtrl.stream,
-        emits(predicate<ParticipantConnectedEvent>(
-          (event) => event.participant.sid == remoteParticipantData.sid,
-        )),
+        emitsInOrder(<Matcher>[
```

Comment:
is this checking to see TrackPublishedEvent is emitted before ParticipantConnectedEvent? I think that's the wrong behavior.